flycheck.el

```diff
@@ -6976,6 +6976,36 @@ Otherwise return the previously used cache directory."
         (or flycheck-haskell-ghc-cache-directory
             (make-temp-file "flycheck-haskell-ghc-cache" 'directory))))
 
+(defun flycheck-haskell--default-directory (checker)
+  "Come up with a sutable default directory for Haskell to run checker in."
+  (pcase checker
+    (`haskell-stack-ghc
+     (or
+      (locate-dominating-file (buffer-file-name) "stack.yaml")
+      (when (executable-find "stack")
+        (let ((stack-dir
+               (replace-regexp-in-string
+                (rx (* (any " \t\n")) eos)
+                ""
+                (shell-command-to-string "stack path --project-root"))))
```

Comment:
Please use process-lines instead of shell-command-to-string.  We should not use shell commands without need.